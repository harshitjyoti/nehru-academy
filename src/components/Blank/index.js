import React from 'react';
import { Result } from 'antd';
const Blank = () => <>
  <Result title="Please execute a search" />
</>
export default Blank;