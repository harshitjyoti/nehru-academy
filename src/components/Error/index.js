import React from 'react';
import { Result, Button } from 'antd';

class ErrorBoundary extends React.Component {
  constructor(props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error) {
    return { hasError: true };
  }

  render() {
    const { hasError } = this.state;
    return <>{hasError ?
      <Result
        status="warning"
        title="There are some problems with your operation."
        extra={
          <Button type="primary" key="console" onClick={() => this.props.history.push('/')}>
            Go To Login
                        </Button>}>
      </Result> :
      this.props.children
    }
    </>
  }
}

export default ErrorBoundary;