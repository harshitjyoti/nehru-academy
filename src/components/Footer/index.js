import React from 'react';
import { Layout, Pagination } from 'antd';

const Footer = ({ count, onSearch }) => <Layout.Footer style={{
    textAlign: 'center',
    position: 'fixed',
    width: '100%',
    bottom: 0
}}>
    <Pagination defaultCurrent={1} total={count} onChange={(page) => onSearch(`page=${page}`)} showSizeChanger={false} />
</Layout.Footer>



export default Footer;