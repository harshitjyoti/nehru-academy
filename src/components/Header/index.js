import React from 'react';
import { Layout, Input, Popover, Anchor, Avatar } from 'antd';
let timer;
const Header = ({ onSearch, isProcessing, user, logout }) => {
  const searchValue = (event) => {
    const { value } = event.target
    clearTimeout(timer)
    timer = setTimeout(() => {
      onSearch(`search=${value}`);
    }, 500)
  }
  return (<Layout.Header style={{ position: 'fixed', zIndex: 1, width: '100%', padding: 0 }}>
    <div style={{ textAlign: "center" }}>
      <Input.Search onChange={searchValue} style={{ width: '50%' }} enterButton={false} size='middle' maxLength='20' placeholder="Search for Planets here !" loading={isProcessing} />
      <Popover content={<Anchor onClick={logout}><Anchor.Link title='Logout' /></Anchor>} title="You Can:">
        <Avatar style={{
          float: 'right',
          margin: 13
        }} size="large">
          {user.charAt(0)}
        </Avatar>
      </Popover>
    </div>
  </Layout.Header>)
}

export default Header;