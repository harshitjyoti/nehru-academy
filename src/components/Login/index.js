import React from 'react';
import { Button, Input, Layout, Card, message } from 'antd';
import { UserOutlined } from '@ant-design/icons';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { loginAction, setUser } from '../../store';

class Login extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: null,
      password: null
    };
  }

  error = () => {
    message.error('Login Failed ! Wrong Credentials !');
    this.props.setUser()

  };

  setUser = (event) => {
    const { id, value } = event.target;
    this.setState({ [id]: value })
  }

  login = () => {
    const { user, password } = this.state;
    this.props.loginAction({ user, password })
  }

  componentDidUpdate() {
    const { isUserFetched, userDetails, history, hasError } = this.props;
    isUserFetched && userDetails && history.push('/planets');
    isUserFetched && !userDetails && this.error()
    hasError && history.push('/error');
  }


  render() {
    const { isProcessing } = this.props;
    const { user, password } = this.state;
    return (
      <Layout.Content style={{
        textAlign: "center",
        top: '50%',
        position: 'absolute',
        left: '50%',
        msTransform: 'translateY(-50%) translateX(-50%)',
        transform: 'translateY(-50%) translateX(-50%)',
        width: '40%'
      }}>
        <div style={{
          textAlign: "center"
        }}>
          <Card title={<><UserOutlined /> Login to Star-Wars</>}>
            <Input disabled={isProcessing} size="large" placeholder="User Name" id='user' prefix={<UserOutlined />} onChange={this.setUser} /><hr />
            <Input.Password disabled={isProcessing} size="large" placeholder="Password" id='password' onChange={this.setUser} /><br /><br />
            <Button disabled={!user || !password} type="primary" onClick={this.login} loading={isProcessing}>
              Login
                </Button>
          </Card>
        </div>
      </Layout.Content>
    );
  }
}

const mapStateToProps = ({ user: { isProcessing, userDetails, hasError, isUserFetched } }) => {
  return {
    isProcessing,
    userDetails,
    hasError,
    isUserFetched
  }
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ loginAction, setUser }, dispatch)
}

export default connect(mapStateToProps, mapDispatchToProps)(Login);