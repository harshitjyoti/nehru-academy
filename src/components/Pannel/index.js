import React from 'react';
import { Collapse, Badge } from 'antd';
import { getColor } from '../../helpers';
import { icons } from '../../constants';
Math.floor(Math.random() * 10)
const getExtra = (population, color) => <Badge style={{ backgroundColor: color }} count={parseInt(population).toLocaleString()} />

const getHeader = (name, color) => <>
{icons[Math.floor(Math.random() * 10)]}
<span style={{ color, fontWeight: 'bold', textDecoration: "underline" }} >{name}</span>
</>
const Pannel = ({ planets }) => {
  return planets.map((item, index) => {
    const color = getColor(item.population)
    return <Collapse key={index}>
      <Collapse.Panel header={getHeader(item.name, color)} key="1" extra={getExtra(item.population, color)}>
        {Object.keys(item).map((i, index) => (
          <span key={index}><b>{i.replace(/_/g, ' ').toUpperCase()}</b> : {item[i]}<hr /></span>
        ))}
      </Collapse.Panel>
    </Collapse>
  }
  )
}

export default Pannel;