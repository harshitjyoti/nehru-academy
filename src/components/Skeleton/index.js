import React from 'react';
import { Skeleton as Skull } from 'antd';

const Skeleton = () => <>
    <Skull active />
    <Skull active />
    <Skull active />
    <Skull active />
</>

export default Skeleton;