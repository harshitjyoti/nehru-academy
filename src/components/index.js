export { default as Blank } from './Blank';
export { default as Footer } from './Footer';
export { default as Header } from './Header';
export { default as Skeleton } from './Skeleton';
export { default as Pannel } from './Pannel';
export { default as Login } from './Login';
export { default as ErrorBoundary } from './Error';
export { default as ErrorComp } from './Error/comp';