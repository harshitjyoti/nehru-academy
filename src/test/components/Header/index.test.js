import React from 'react';
import { Header } from '../../../components';
import renderer from 'react-test-renderer';

test('SnapShot test', () => {
  const component = renderer.create(
    <Header onSearch={()=> true} isProcessing={false} user={'Harhsit'} logout={()=> true}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});