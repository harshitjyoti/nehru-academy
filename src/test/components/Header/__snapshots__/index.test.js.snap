// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnapShot test 1`] = `
<header
  className="ant-layout-header"
  style={
    Object {
      "padding": 0,
      "position": "fixed",
      "width": "100%",
      "zIndex": 1,
    }
  }
>
  <div
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <span
      className="ant-input-search ant-input-affix-wrapper"
      onMouseUp={[Function]}
      style={
        Object {
          "width": "50%",
        }
      }
    >
      <input
        className="ant-input"
        maxLength="20"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Search for Planets here !"
        style={null}
        type="text"
        value=""
      />
      <span
        className="ant-input-suffix"
      >
        <span
          aria-label="search"
          className="anticon anticon-search ant-input-search-icon"
          onClick={[Function]}
          role="img"
          tabIndex={-1}
        >
          <svg
            aria-hidden="true"
            className=""
            data-icon="search"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
            />
          </svg>
        </span>
      </span>
    </span>
    <span
      className="ant-avatar ant-avatar-lg ant-avatar-circle"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "float": "right",
          "margin": 13,
        }
      }
    >
      <span
        className="ant-avatar-string"
        style={
          Object {
            "opacity": 0,
          }
        }
      >
        H
      </span>
    </span>
  </div>
</header>
`;
